XKCDY/app
# Add Shuffle Button to Comics Grid

## Feature Description
Adds an option to view a random comic directly from the comics grid.

## Changelist
- Third button added to floating buttons row (see attached screenshots)
    - On iOS, button is centered horizontally in row
    - On iPadOS, button is grouped with other floating buttons
- Button press opens comic pager already shuffled
- Floating buttons preview updated with test response

## Testing Steps
- Open XKCDY to the comics grid, confirming shuffle button is visible matching the above specs. 
- If not already available, add 2 or more comics to the Favorites tab.
- Press the shuffle button in the All tab, confirming that a random comic is displayed.
- Press the shuffle button in the Favorites tab, confirming that only a random favorite comic is displayed.

## E2E Criteria
- Settings and search buttons work as intended when pressed
- All three buttons work and display as intended in both portrait and landscape mode
- Pressing on a comic from the comics grid works as intended, before and after shuffling from the grid
- All actions from the comic overlay work as intended for the target comic
- Returning to the comics grid from a random comic behaves the same as from a non-random comic

## Issues Noticed During Testing
- Rapidly switching between the All and Favorites tab can crash the application. Reproducible in stable version without the above changes.

## Screenshots
![Shuffle Button - iPhone 8 Plus](https://user-images.githubusercontent.com/14227901/91667789-b2aafe00-eacc-11ea-9189-365cd15169c9.png)



![Shuffle Button - iPad Pro (12 9-inch) (4th generation)](https://user-images.githubusercontent.com/14227901/91667816-e2f29c80-eacc-11ea-9bd0-f5f9fe601d4f.png)
